libxrdp/xrdp_jpeg_compress.c

```diff
@@ -265,8 +265,8 @@ my_term_destination(j_compress_ptr cinfo)
 
 /*****************************************************************************/
 static int APP_CC
-jp_do_compress(char *data, int width, int height, int bpp, int quality,
-               char *comp_data, int *comp_data_bytes)
+jp_do_compress(JOCTET *data, int width, int height, int bpp, int quality,
```

Comment:
If cleaning up this code, you may also want to add `const` to the pointer (unless it leads to a cascade of changes, in which case it would need to be a separate patch).